### DISCUSSION TITLE
Deflection of outer skull & scalp boundaries in initial registration and segmentation

### DISCUSSION THREAD

**Discussion by divyarajasekharan** [simnibs/simnibs#200] [7.Creating head models]
I've been observing that for a number of our subjects,  the initial registration deflects the skull & scalp boundaries inwards at the front of the head:

![image](https://github.com/simnibs/simnibs/assets/91343323/1acb8a3a-e1f6-41ae-bf5b-6a8ba99c625a)

This causes the anterior part of the scalp & skull to be masked out in the T1_upsampled image & affects the resulting segmentation:

![image](https://github.com/simnibs/simnibs/assets/91343323/ae3d57c8-0928-42f7-9b84-cec64d32ac40)

I'm not sure whether varying the background_mask_sigma, background_mask_threshold, or perhaps the bias_kernel_width in the settings file might help, or if this is error originates fundamentally with the initial registration (which seems likely). This issue has also begun occurring more frequently when I began using the settings linked in this [discussion thread](https://github.com/simnibs/simnibs/discussions/131#discussioncomment-5636749) in order to improve the GM segmentation. Please let me know what you advise, and thank you very much again for all your wonderful & prompt assistance.

**Comment by jduemose** [comment]
Hi

I am suspecting a poor affine registration of the template to be the culprit here (as you also alluded to). If you choose "coregistered template" in the overlay section of the `charm_report.html`, do you see anything that might explain this result? I could imagine the scaling being a bit off. If the problem seems to be with the initial affine registration, then you might want to take a look [here](https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_affine_registration.html#bad-scaling) which shows how to edit the settings file to alleviate these kinds of issues.

Best,
Jesper

**Comment by divyarajasekharan** [comment]
Hi Jesper,

I can't believe I missed that! You're completely right that this is a case of bad scaling for every subject for which we're seeing this scalp/skull indentation. I'm running these through again with a higher upper bound on the affine scales in the x-y plane. I will keep you posted about the results, but I expect them to be much better! Thank you so much!

**Reply by divyarajasekharan** [reply]
I've been playing around with the affine scales without much success in improving the initial registration. Are there any parameters in the settings file that could be modified to prevent tissues from being masked into the background in the T1_upsampled image, shown as an overlay in red below? This would make it easier to manually correct the segmentation, if nothing else. 
 
![image](https://github.com/simnibs/simnibs/assets/91343323/c1cbf7f8-c2a0-46c6-8a57-e627bcdf2d46)
 
Thank you!

**Reply by jduemose** [reply]
Hi, can you share an image of how the initial coreregistered template looks when overlaid on the structural image and what parameter settings you tried? It might be that the template is a really bad fit to this particular subject.

**Comment by dmitrishastin** [comment]
Hi, 

Unfortunately, I am having the same issue (in about 30% of my 120 subjects). My cohort only has T1w images. I am using SimNIBS 4.0.0. A significant proportion of my cohort has atrophy, but there does not seem to be an obvious correlation between atrophy and this issue happening (I have also tried suggestions in #131 for completeness). I have tried tweaking settings quite aggressively in other ways too, but no luck. I have tried forcing affine registration that I get with ANTs and, without going into much detail, the outcome was similar. Looking at report html, the template overlay does look slightly off in places (like sections of the skull, similar to this post) but the position is good overall; intracranial tissues appear to be segmented well. My next option is to try non-affine registration to MNI followed by inverse warp of the meshes as in #156 ; the fall-back plan is manual re-labelling but that would of course be really labour intensive. Just wanted to check if you had any alternative suggestions at all? Would appreciate any help!

**Reply by oulap** [reply]
Hi,

Sounds like you've tried quite a bit of things. Is it possible for you to share a screenshot of the problematic T1s? You can also send that to support@simnibs.org if you do not want to post it here. 

Sometimes, depending on the T1 scan settings, the fatty tissues can appear bright or dark. I tweaked charm to prefer the latter, as that's the "official" simnibs recommended setting. It's possible to tweak the parameters the other way as well, but I would probably need an example scan for that.
